src/nvim/path.c

```diff
@@ -412,7 +412,7 @@ void add_pathsep(char *p)
 char *FullName_save(char *fname, bool force)
   FUNC_ATTR_NONNULL_RET FUNC_ATTR_MALLOC
 {
-  if (fname == NULL) {
+  if (fname == NULL || STRLEN(fname) > MAXPATHL) {
```

Comment:
On line 421 the FAIL result means `vim_FullName()` truncated the string (as it should). But then in the `else` branch we just blindly copy the string without truncating it. Instead we should truncate it.

    new_fname = xstrndup(fname, len);